drivers/video/gc2145.c

```diff
@@ -901,28 +905,65 @@ static int gc2145_set_output_format(const struct device *dev, int output_format)
 	return 0;
 }
 
-static int gc2145_set_resolution(const struct device *dev, uint16_t img_width, uint16_t img_height)
+static int gc2145_set_resolution(const struct device *dev, enum resolutions res)
 {
 	int ret;
 	const struct gc2145_config *cfg = dev->config;
 
-	uint16_t win_w = UXGA_HSIZE + 16;
-	uint16_t win_h = UXGA_VSIZE + 8;
-	uint16_t c_ratio = 1;
-	uint16_t r_ratio = 1;
-	uint16_t x = (((win_w / c_ratio) - UXGA_HSIZE) / 2);
-	uint16_t y = (((win_h / r_ratio) - UXGA_VSIZE) / 2);
-	uint16_t win_x = ((UXGA_HSIZE - win_w) / 2);
-	uint16_t win_y = ((UXGA_VSIZE - win_h) / 2);
+	uint16_t w = fmts[res].width_min;
+	uint16_t h = fmts[res].height_min;
```

Comment:
Looks like an oversight - there's already a check and a safe assignment below.
```suggestion
	uint16_t w;
	uint16_t h;
```